heapq.c

```diff
@@ -44,7 +45,7 @@ static void heapify_up(struct heapq_t *hq)
     int idx = hq->size - 1;
     int parent_idx = heapq_parent_idx(idx);
     /* keep "repearing" heap as long as parent is greater than child */
-    while (parent_idx >= 0 && hq->cmp(hq->items[parent_idx], hq->items[idx])) {
+    while (parent_idx >= 0 && hq->cmp(hq->items[parent_idx], hq->items[idx]) > 0) {
```

Comment:
what does adding `> 0` do here?

https://en.wikipedia.org/wiki/Comma_operator

Reply:
New compare function, not a boolean but rather an int so we can check for equality as well.

Reply:
We are comparing if the statement is larger than 0.